### DISCUSSION TITLE
redirect from `/:user/:repo/issues/:num` to `/:user/:repo/pull/:num` remove in new issues?

### DISCUSSION THREAD

**Discussion by trentm** [community/community#8623] [Projects and Issues] [answered]
It used to be (as recently as yesterday for me) that browsing to `/:user/:repo/issues/:num` used to redirect to `/:user/:repo/pull/:num` if `:num` happened to be a PR.  For example `https://github.com/elastic/apm-agent-nodejs/issues/2001` redirects to `https://github.com/elastic/apm-agent-nodejs/pull/2001`.

Was there a recent change to Issues such that there is no such redirect now **if one is logged in and enabled for GitHub's new Issues**?

I *do* get the redirect when I try in a private browsing window (or separate browser) where I am not logged into GitHub (example request `x-github-request-id: DD0E:192B:1A8DC54:2C55C9C:61B27668` if you are looking in logs).  But if I'm logged in, I do not get the redirect, I get an HTTP 404 (e.g. `x-github-request-id: DA5D:7CB7:46FFE9:DADC55:61B275BD`)

**Comment by joshtriplett** [comment]
In particular, this has caused issues for project forums and project communication channels that want to automatically link `#NNNNN` or `project#NNNNN` to an issue by just textually generating a URL, without looking up the number to see if it's an issue or PR.

**Comment by kaste** [comment]
The other way around, t.i. using `/:user/:repo/pull/:num`, it still redirects to `.../issues/:num` if necessary.  🤷

**Reply by trentm** [reply]
@kaste Thanks. That's a sufficient workaround for my case -- what Josh described (`want to automatically link #NNNNN or project#NNNNN to an issue by just textually generating a URL`).

**Comment by azenMatt** [comment]
Hi everyone - we have rolled this change back for the moment, 

Would you be able to expand more on your workflows and where it is helpful for you to have a single reference to either an issue or PR?

**Reply by ericrbg** [reply]
Zulip chats usually link either issues or PRs using the #<id> syntax; this is far easier to use if one type of link redirects to the other.

**Reply by HonkingGoose** [reply]
Quote [^quote] from one of the maintainers of the `semantic-release` package which helps make automated release-notes:

> for some context, [`@semantic-release/release-notes-generator`](https://github.com/semantic-release/release-notes-generator) uses [`conventional-changelog-writer`](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-writer) to generate the notes.
> 
> i found [an issue](https://github.com/conventional-changelog/conventional-changelog/issues/655) there where someone was asking about the notes always using `issues` instead of `pull`, which confirms that this is intentional behavior that relies on the redirects, in case that needs confirmation.

If the `semantic-release` package or the `conventional-changelog-writer` package need to figure out if the change is coming from an issue or a PR it will probably be extra work for them.

This change broke the URLs in the auto-generated changelogs for the Renovate project, as we use `semantic-release` to create our changelogs.

I hope this is helpful to understand why some people/bots rely on this behavior right now.

[^quote]: https://github.com/renovatebot/renovate/issues/13038#issuecomment-990215955

**Reply by Chew** [reply]
The worst part about it isn't the future, it's maintaining backwards compatibility.

Even if people could go through and fix future links, all prior links to PRs from the past, that link to /issues, would be broken, sometimes unfixable as well.

It is far easier for us to link to the "issue/pr" especially since there's no easy way to distinguish them in the API (to my knowledge, we can't even get a PR count in the REST API). I obviously don't know the internals of GitHub, or how much load the redirect causes, but reverting it with no warning is a huge change no one was really ready for, so much so people thought it was a bug.

**Reply by kaste** [reply]
I'm maintaining GitSavvy, a git client.  We parse commit messages and link back to the GitHub pages.  People usually refer to issues or prs (from the same project) just using `#123` notation.  For example: `We did something like this in #2311 for ...`, `Fixes #123`, `See #4566`.  `Merge pull request #1516 from ...`.  It is conventional (in natural language) to not mention and distinguish between "number types", t.i. is it an issue or a pr?  Of course, for this use case, we *could* ask an API to tell us the correct html url given a `#123` number but this is surely not faster than the redirect a browser enjoys doing anyway. 

![image](https://user-images.githubusercontent.com/8558/145874743-2e1a7807-4b51-4167-bd95-cfc6988fd6b8.png)

**Reply by azenMatt** [reply]
Thanks everyone for your input here, this should all be back working for you as usual. But please let me know if you are still seeing any unexpected behaviour from this.

**Comment by azenMatt** [comment, marked as answer]
Hi everyone, apologies for this disruption this caused, we have rolled back the change that caused this
